controllers/cron_rollout_helpers.go

```diff
@@ -174,41 +175,36 @@ func CreateDisruptionFromTemplate(ctx context.Context, cl client.Client, scheme
 }
 
 // getScheduledTimeForDisruption returns the scheduled time for a particular disruption.
-func getScheduledTimeForDisruption(disruption *chaosv1beta1.Disruption) (*time.Time, error) {
+func getScheduledTimeForDisruption(log *zap.SugaredLogger, disruption *chaosv1beta1.Disruption) time.Time {
 	timeRaw := disruption.Annotations[ScheduledAtAnnotation]
 	if len(timeRaw) == 0 {
-		return nil, nil
+		return time.Time{}
 	}
 
 	timeParsed, err := time.Parse(time.RFC3339, timeRaw)
 	if err != nil {
-		return nil, err
+		log.Errorw("unable to parse schedule time for child disruption", "err", err, "disruptionName", disruption.Name)
+		return time.Time{}
 	}
 
-	return &timeParsed, nil
+	return timeParsed
 }
 
 // getMostRecentScheduleTime returns the most recent scheduled time from a list of disruptions.
-func GetMostRecentScheduleTime(log *zap.SugaredLogger, disruptions *chaosv1beta1.DisruptionList) *time.Time {
-	var mostRecentScheduleTime *time.Time
-
-	for _, disruption := range disruptions.Items {
-		scheduledTimeForDisruption, err := getScheduledTimeForDisruption(&disruption)
-		if err != nil {
-			log.Errorw("unable to parse schedule time for child disruption", "err", err, "disruption", disruption.Name)
-			continue
-		}
-
-		if scheduledTimeForDisruption != nil {
-			if mostRecentScheduleTime == nil {
-				mostRecentScheduleTime = scheduledTimeForDisruption
-			} else if mostRecentScheduleTime.Before(*scheduledTimeForDisruption) {
-				mostRecentScheduleTime = scheduledTimeForDisruption
-			}
-		}
-	}
-
-	return mostRecentScheduleTime
+func GetMostRecentScheduleTime(log *zap.SugaredLogger, disruptions *chaosv1beta1.DisruptionList) time.Time {
+	length := len(disruptions.Items)
+	if length == 0 {
+		return time.Time{}
+	}
+
+	sort.Slice(disruptions.Items, func(i, j int) bool {
```

Comment:
The initial linear fashion of finding the most recent time should be faster and more efficient than sorting the items and picking out the last item.

Reply:
true. Its just a hunch that the previous code is buggy. 

I personally find the sorting easier to read. In practice, this list is always an almost-sorted list. I'd rather pay a small fee to ensure it's really sorted, to avoid wrong assumption during debugging.

Reply:
I'm not sure how it could be buggy, it seems straightforward. The sorting does seem cleaner, but you can also encapsulate a lot of the code in the previous code into smaller functions to also make it cleaner. The fee is dependent on the # of disruptions in that list, if there are 100 disruptions, then the sorting is ~6X slower, which may not be a small fee especially if it grows. Also, we dont need disruptions items to be sorted from my understanding besides in this function where you grab the last item from the sorted list, unless there is another place where having this list be sorted again makes sense.